Submitted By:            Bruce Dubbs <bdubbs_at_linuxfromscratch_dot_org>
Date:                    2012-04-30
Initial Package Version: 5.3.3
Upstream Status:         Unknown
Origin:                  Upstream
Description:             Fixes video4linux scanner backend build problems.

diff -Naur sane-backends-1.0.22.orig/backend/v4l.h sane-backends-1.0.22/backend/v4l.h
--- sane-backends-1.0.22.orig/backend/v4l.h	2010-12-01 17:49:58.000000000 -0600
+++ sane-backends-1.0.22/backend/v4l.h	2012-04-30 14:03:53.000000000 -0500
@@ -29,6 +29,8 @@
 #ifndef v4l_h
 #define v4l_h
 
+#include <libv4l1-videodev.h> 
+
 /* Kernel interface */
 /* Only the stuff we need. For more features, more defines are needed */
 
@@ -47,6 +49,7 @@
 #define VID_TYPE_MJPEG_DECODER	4096	/* Can decode MJPEG streams */
 #define VID_TYPE_MJPEG_ENCODER	8192	/* Can encode MJPEG streams */
 
+#if 0
 struct video_capability
 {
 	char name[32];
@@ -102,9 +105,10 @@
 /* bitmap is 1024x625, a '1' bit represents a clipped pixel */
 #define VIDEO_CLIPMAP_SIZE	(128 * 625)
 };
-
+#endif
 #define VIDEO_MAX_FRAME		32
 
+#if 0 
 struct video_mbuf
 {
 	int	size;		/* Total memory to map */
@@ -132,6 +136,7 @@
 #define VIDEO_TYPE_CAMERA	2	
 	__u16 norm;			/* Norm set by channel */
 };
+#endif
 
 #define VIDIOCGCAP		_IOR('v',1,struct video_capability)	/* Get capabilities */
 #define VIDIOCGCHAN		_IOWR('v',2,struct video_channel)	/* Get channel info (sources) */
